### PR TITLE
Update variable used in Dryrun job decision

### PR DESCRIPTION
**Pull request by qduk**
This changes the variable looked at for dryrun to be `self.sync.dry_run` instead of `self.dryrun`.

I can pull out the format changes if need be.

### CODE REVIEW

File: nautobot_ssot/tests/test_jobs.py [hunk 1]
```diff
@@ -113,9 +139,6 @@ def test_calculate_diff_fail_diff_save_too_large(self):
         self.job.logger.warning = Mock()
         self.job.source_adapter.diff_to().dict.return_value = {}
         self.job.calculate_diff()
-        self.job.logger.warning.assert_any_call(
```
Comment (chadell) [inline nautobot_ssot/tests/test_jobs.py:116]: why is this no longer valid?

**Review by chadell — APPROVED** [review]
LGTM

File: changes/548.fixed [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed SSoT jobs not respecting DryRun variable
```
Comment (gsnider2195) [inline changes/548.fixed:1]:
```suggestion
Fixed SSoT jobs not respecting DryRun variable.
```

**Comment by gsnider2195** [comment]
Are you still using black for formatting? It looks like your autoformatter has defaulted to 100 character line-width which is creating a bunch of noise. There's a bug in the `ruff-format` ci check that causes it to always pass but ruff formatting is actually failing if you look at https://github.com/nautobot/nautobot-app-ssot/actions/runs/10890864505/job/30220509988?pr=549

File: nautobot_ssot/tests/test_jobs.py (hunk 1 above)
Comment (qduk) [reply nautobot_ssot/tests/test_jobs.py:116]: Thanks for catching that Christian!

**Comment by qduk** [comment]
@gsnider2195  I've gone and fixed ruff. I had a unittest failing as well but I've gone and resolved that.

**Comment by gsnider2195** [comment]
> @gsnider2195 I've gone and fixed ruff. I had a unittest failing as well but I've gone and resolved that.

Ruff is still complaining about one file:

> Would reformat: nautobot_ssot/jobs/base.py

**Comment by qduk** [comment]
> > @gsnider2195 I've gone and fixed ruff. I had a unittest failing as well but I've gone and resolved that.
> 
> Ruff is still complaining about one file:
> 
> > Would reformat: nautobot_ssot/jobs/base.py

Now it's "ruffed". Thanks Gary!

**Review by jdrew82 — APPROVED** [review]
LGTM!

**Review by jdrew82 — APPROVED** [review]
LGTM!

**Pull request merged**